prusti-viper/src/encoder/backtranslation.rs

```diff
@@ -0,0 +1,457 @@
+                tcx,
+                is_pure,
+                disc_info,
+            })
+        } else {
+            None
+        }
+    }
+}
+
+
+pub fn backtranslate<'tcx>(
+    encoder: &Encoder,
+    def_id: ProcedureDefId,
+    silicon_counterexample: Option<SiliconCounterexample>,
+) -> Counterexample {
+    let opt_backtranslator = Backtranslator::new(encoder, def_id, silicon_counterexample);
+    if let Some(backtranslator) = opt_backtranslator {
+        //get all the needed information from the mir
+        let var_debug_info = backtranslator.mir.var_debug_info.clone();
+        let local_variable_manager = LocalVariableManager::new(&backtranslator.mir.local_decls);
+        let arg_count = backtranslator.mir.arg_count;
+
+        //optimally (at a later stage) we would use the "main" counterexample 
+        //from silicon, the one not associated with any label, because it contains
+        //the values of the function when it fails. But currently
+        //most values can not be obtained there because they're folded
+        let last_label: Option<&String> = backtranslator
+            .silicon_counterexample
+            .label_order.last();
+        //optimally this label would just be "old", but as of now mostly the values
+        //are not available at this point
+        let old_label = String::from("l0");
+
+
+
+        //to be processed:
+        let mut args_to_process: Vec<(String, MultiSpan, String, Ty)> = vec![];
+        let mut entries_to_process: Vec<(String, MultiSpan, String, Ty)> = vec![];
+
+        for vdi in var_debug_info{
+            let rust_name = vdi.name.to_ident_string();
+            let span = vdi.source_info.span;
+            let multi_span = MultiSpan::from_span(span);
+            let local: mir::Local = if let mir::VarDebugInfoContents::Place(place) = vdi.value {
+                if let Some(local) = place.as_local() {
+                    local
+                } else {
+                    continue;
+                }
+            } else {
+                continue;
+            };
+            let index = local.index();
+            let var_local = Local::from(local);
+            let typ = local_variable_manager.get_type(var_local);
+            let vir_name = local_variable_manager.get_name(var_local);
+            entries_to_process.push((rust_name.clone(), multi_span.clone(), vir_name.clone(), typ));
+            
+            //if index indicates it is an argument
+            if index > 0 && index <= arg_count {
+                args_to_process.push((rust_name, multi_span, vir_name, typ))
+            }
+        }
+
+        //for the return type:
+        let return_local = Local::from(mir::Local::from_usize(0));
+        //make sure
+        let result_to_process = if local_variable_manager.is_return(return_local){
+            //open question: what would be the right span for return type? None?
+            let rust_name = String::from("result");
```

Comment:
```suggestion
            let rust_name = "result".to_string();
```
(and likewise in other places; do you really need a `String` also?)